girder/web_client/src/views/widgets/BrowserWidget.js

```diff
@@ -238,6 +268,46 @@ var BrowserWidget = View.extend({
                         _.escape(invalidInputElement) + '<br>' + _.escape(invalidSelectedModel));
                 }
             });
+    },
+
+    /**
+     * If we have a defaultSelectedResource we need the root item for the hiearachyWidget
+     * This will calcualte what the root item should be if one hasn't been supplied
+     */
+    _calculateDefaultSelectedRoot: function () {
+        // If we are are only using folders, the root is the defaultselectedResource then
+        if (!this.selectItem) {
+            this.root = this.defaultSelectedResource;
+            return;
+        }
+        // We need to calculate the parent model of the default.selectedResoure when selectItem is true
+        const modelTypes = {
+            folder: FolderModel,
+            collection: CollectionModel,
+            user: UserModel
+        };
+        let modelType = 'folder'; // folder type by default, other types if necessary
+        let modelId = null;
+        // If it is an item it will have a folderId associated with it as a parent item
+        if (this.defaultSelectedResource.get('folderId')) {
+            modelId = this.defaultSelectedResource.get('folderId');
+        } else if (this.defaultSelectedResource.get('parentCollection')) {
+            // Case for providing a folder as the defaultSelectedResource but want the user to select an item
+            // folder parent is either 'user' | 'folder' | 'collection', most likely folder though
+            modelType = this.defaultSelectedResource.get('parentCollection');
+            modelId = this.defaultSelectedResource.get('parentId');
```

Comment:
Can you add to the tests to cover this condition, too?

Reply:
New tests added should cover this specific case as well.